airflow/providers/amazon/aws/hooks/logs.py

```diff
@@ -25,6 +25,13 @@
 
 from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
 
+# From AWS team the correct way to check for end of stream is that if the value of nextForwardToken is same
+# in subsequent calls
+# The issue with this approach is, it can take a huge amount of time (e.g. 20 seconds) to retrieve logs
+# As an intermediate solution, we decided to stop fetching logs if 3 consecutive responses are empty
```

Comment:
if that happens what user is expected to do / how would user know that this is the issue ?

Reply:
He would not be able to know. Before #20814, we were stopping fetching logs after one response from CW is empty. After #20814 being introduced, we now wait that the next token is the same in 2 consecutive responses. We now shoot in the middle to satisfy a bit of the two worlds. This fix is very empirical based on user experiences. They get a very frustrated experience where, sometimes, they need to wait 20 seconds to retrieve logs